### PR TITLE
hit: add support for reflowing long strings

### LINKED ISSUE #10889: CIVET: 'Minimum gcc version' failure
Testing 901dc26

[Minimum gcc version](https://www.moosebuild.org/job/160477/) : **Failed**

View the results [here](https://www.moosebuild.org/job/160477/).

### PR DESCRIPTION
**Pull request by rwcarlsen**
This does not add any formal auto-formatting support, but is helping to lay the groundwork for a canonical moose input file style enforceable with auto-formatting.

Also fix handling of blank lines and inline comments.  PR #10883 broke blank line and comment handling.  The new consecutive string literal syntax was causing blank line tokens after the last string literal to be "eaten" as part of the string field node by the parser. This adds tests that should prevent such regressions in addition to fixing the issue. Fixes #10889

### CODE REVIEW

**Comment by brianmoose** [comment]
The peacock failures are due to hit now parsing:
```
[u]
  vars = 'u'
  # PETSc options for this subsolver
  # A prefix will be applied, so just put the options for this subsolver only
  petsc_options_iname = '-pc_type -ksp_type'
  petsc_options_value = '     hypre preonly'
[]
```

into

```
[u]
  vars = 'u' # PETSc options for this subsolver
  # A prefix will be applied, so just put the options for this subsolver only
  petsc_options_iname = '-pc_type -ksp_type'
  petsc_options_value = '     hypre preonly'
[]
```

File: framework/contrib/hit/main.cc [hunk 1]
```diff
@@ -267,6 +279,8 @@ format(int argc, char ** argv)
     return 1;
   }
 
+  int maxlen = std::stoi(flags.val("maxlen"));
```
Comment (permcody) [inline framework/contrib/hit/main.cc]: What about 5elephant? How many times are we going to write this same bug?

File: framework/contrib/hit/parse.cc [hunk 2]
```diff
@@ -17,7 +19,16 @@
 namespace hit
 {
 
-const std::string indentString = "  ";
+// returns the type of quoting used on string s (i.e. " or ') or an empty string otherwise.
+std::string
```
Comment (permcody) [inline framework/contrib/hit/parse.cc]: Consider a return type of `char`.

File: framework/contrib/hit/parse.cc [hunk 3]
```diff
@@ -418,11 +429,43 @@ Field::path()
...
+    }
+    if (pos < unquoted.size())
+    {
+      if (pos > 0)
+        s += "\n" + strRepeat(" ", prefix_len);
+      s += quote + unquoted.substr(pos, std::string::npos) + quote;
+    }
+  }
+  else
+    s += _val;
```
Comment (permcody) [inline framework/contrib/hit/parse.cc]: This routine is not exactly clear. How about a few comments about what's going on here.

**Review by permcody — CHANGES_REQUESTED** [review]

File: framework/contrib/hit/main.cc (hunk 1 above)
Comment (rwcarlsen) [reply framework/contrib/hit/main.cc]: This is sort of just a play utility I've used for dinking around testing/experimenting with hit stuff - not any sort of official binary meant to be used by others.

File: framework/contrib/hit/parse.cc (hunk 2 above)
Comment (rwcarlsen) [reply framework/contrib/hit/parse.cc]: I did, but the contexts where I call this function often need a std::string too - so I just decided on string.  This is an internal function that isn't an exposed API.

File: framework/contrib/hit/parse.cc (hunk 3 above)
Comment (rwcarlsen) [reply framework/contrib/hit/parse.cc]: Done.

File: framework/contrib/hit/parse.cc (hunk 2 above)
Comment (permcody) [reply framework/contrib/hit/parse.cc]: Thanks!

**Review by permcody — APPROVED** [review]

**Comment by brianmoose** [comment]
@milljm noticed that `outputs/format.json` test recently slowed down significantly.
Doing a git bisect shows that this PR (9603d134d684e32b16719d88d833d7977c283a91)
is where the slowdown starts.
The test was ~10 seconds before then ~40 after on my mac.
To be fair, this test uses Hit to parse the output of `--dump` to it is a huge input file.

**Pull request merged**